lib/Form/Processor/PolicyFormProcessor.php

```diff
@@ -47,7 +48,17 @@ public function processUpdatePolicy(FormActionEvent $event)
             list($module, $function) = explode('|', $data->moduleFunction);
             $data->module = $module;
             $data->function = $function;
-            $this->roleService->addPolicyByRoleDraft($data->roleDraft, $data);
+            $initialRoleDraft = $data->roleDraft;
+            $updatedRoleDraft = $this->roleService->addPolicyByRoleDraft($initialRoleDraft, $data);
+
+            $initialPoliciesById = $this->getPoliciesById($initialRoleDraft);
+            $updatedPoliciesById = $this->getPoliciesById($updatedRoleDraft);
+            foreach ($updatedPoliciesById as $policyId => $policyDraft) {
+                if (!isset($initialPoliciesById[$policyId])) {
+                    $data->setPolicyDraft($policyDraft);
```

Comment:
In my test, this fails as `$policyDraft` is a `Policy`, not a draft. Am I missing something?